index.js

```diff
@@ -0,0 +1,57 @@
+module.exports = AsMVT;
+
+const tilelive = require("@mapbox/tilelive");
+const url = require("url");
+const zlib = require("zlib");
+const { Pool } = require("pg");
+
+function AsMVT(uri, callback) {
+  var parsedUrl = url.parse(uri, true);
+  const functionName = parsedUrl.query.function || "gettile";
+  this._query = "select " + functionName + "($1, $2, $3)";
+  this.pool = new Pool({
+    user: parsedUrl.query.user || process.env.PGUSER,
+    host: parsedUrl.query.host || process.env.PGHOST,
+    database: parsedUrl.query.database || process.env.PGDATABASE,
+    password: parsedUrl.query.password || process.env.PGPASSWORD,
+    port: parsedUrl.query.port || process.env.PGPORT
+  });
+  callback(null, this);
+}
+
+AsMVT.prototype.getTile = function(z, x, y, callback) {
+  this.pool.query(this._query, [z, x, y], (err, res) => {
+    if (err) {
+      return callback(err);
+    }
+    return zlib.gzip(res.rows[0][0], function(err, pbfz) {
+      if (err) {
+        return callback(err);
+      }
+      return callback(null, pbfz, {
+        "Content-Encoding": "gzip",
+        "Content-Type": "application/x-protobuf"
+      });
+    });
+  });
+};
+
+AsMVT.prototype.getInfo = function(callback) {
+  callback(null, {
+    name: "tilelive_AsMVT tileset",
```

Comment:
This info object should probably be user-configurable, but I'm not sure what the best API for that might be 🤔

Reply:
It should be, but we have no need for it right now, so im not going to bother.